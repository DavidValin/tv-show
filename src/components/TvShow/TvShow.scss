@import '../../shared.scss';

.TvShow {
  display: flex;
  @include fadeIn();

  @media (min-width: $screen-md-min) {
    padding: 2em;
  }
  @media (max-width: $screen-sm-max) {
    flex-wrap: wrap;
  }

  &__cover {
    img {
      width: 100%;
    }
  }

  @media (min-width: $screen-md-min) {
    &__cover,
    &__details {
      width: 50%;
    }
  }
  @media (max-width: $screen-sm-max) {
    &__cover {
      width: calc(100% - 2em);
    }
    &__details {
      width: 100%;
    }
  }

  &__cover,
  &__details {
    margin: 1em;
    &__episodes {
      &__heading {
        font-weight: 600;
        border-bottom: 2px solid black;
        padding-bottom: 1em;
      }
      &__episody {
        padding: 1em;
        border: 1px solid #eaeaea;
        display: block;

        &:hover {
          background: #efefef;
        }
      }
    }
  }
}
