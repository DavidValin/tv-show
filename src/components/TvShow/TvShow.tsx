import React from 'react';
import { Link } from 'react-router-dom';
import './TvShow.scss';
import { ITvShow } from '../../state/initial_state';

export interface IParams {
  tvShow: ITvShow
}

const TvShow = (props:IParams) => (
  <div className='TvShow'>
    <div className='TvShow__cover'>
      <img src={props.tvShow.coverImg.original} alt={props.tvShow.title} />
    </div>
    <div className='TvShow__details'>
      <h2>{props.tvShow.title}</h2>
      <p>{props.tvShow.description}</p>
      <div className='TvShow__details__episodes'>
        <h3 className='TvShow__details__episodes__heading'>
          Episodes
        </h3>
        {props.tvShow.episodes.map((episody, i) => (
          <Link key={i} to={`/shows/${props.tvShow.id}/episodes/${episody.id}`} className='TvShow__details__episodes__episody'>
            {episody.name}
          </Link>
        ))}
      </div>
    </div>
  </div>
)

export default TvShow;

