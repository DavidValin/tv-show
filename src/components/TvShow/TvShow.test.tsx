import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import TvShow from './TvShow';
import { tvShowTestData } from './TvShow.testData';

it('should match the snapshot', () => {
  const tree = renderer.create(<Router><TvShow tvShow={tvShowTestData.tvShow} /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});