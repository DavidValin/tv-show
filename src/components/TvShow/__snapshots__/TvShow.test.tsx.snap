// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="TvShow"
>
  <div
    className="TvShow__cover"
  >
    <img
      alt="A test Tv Show"
      src="http://static.tvmaze.com/uploads/images/original_untouched/53/132629.jpg"
    />
  </div>
  <div
    className="TvShow__details"
  >
    <h2>
      A test Tv Show
    </h2>
    <p>
      Lorem Ipsum is slechts een proeftekst uit het drukkerij- en zetterijwezen. Lorem Ipsum is de standaard proeftekst in deze bedrijfstak sinds de 16e eeuw, toen een onbekende drukker een zethaak met letters nam en ze door elkaar husselde om een font-catalogus te maken. Het heeft niet alleen vijf eeuwen overleefd maar is ook, vrijwel onveranderd, overgenomen in elektronische letterzetting. Het is in de jaren '60 populair geworden met de introductie van Letraset vellen met Lorem Ipsum passages en meer recentelijk door desktop publishing software zoals Aldus PageMaker die versies van Lorem Ipsum bevatten.
    </p>
    <div
      className="TvShow__details__episodes"
    >
      <h3
        className="TvShow__details__episodes__heading"
      >
        Episodes
      </h3>
      <a
        className="TvShow__details__episodes__episody"
        href="/shows/1/episodes/657308"
        onClick={[Function]}
      >
        Escape from Monster Island
      </a>
      <a
        className="TvShow__details__episodes__episody"
        href="/shows/1/episodes/657309"
        onClick={[Function]}
      >
        Princess Buttercup
      </a>
    </div>
  </div>
</div>
`;
